sklearn/utils/_array_api.py

```diff
@@ -626,6 +640,40 @@ def get_namespace_and_device(*array_list, remove_none=True, remove_types=(str,))
         return xp, False, arrays_device
 
 
+def move_to_namespace_and_device(*arrays_to_move, ref):
+    """Helper to implement the 'y follows X' rule.
+
+    Convert arrays to the namespace and device of ``ref``.
+
+    When ``ref`` is not an array api array, the inputs are returned unchanged.
+    """
+    xp, is_array_api = get_namespace(ref)
+    if not is_array_api:
+        return arrays_to_move
+
+    device_ = device(ref)
+
+    new_arrays = []
+    for array in arrays_to_move:
+        if array is None or isinstance(array, (numbers.Number, str)):
+            new_arrays.append(array)
+            continue
+        array_xp, _, array_device = get_namespace_and_device(array)
+        if array_xp == xp and array_device == device_:
+            new_arrays.append(array)
+            continue
+        try:
+            new_arrays.append(xp.asarray(array, device=device_))
+            continue
+        except Exception:
+            # direct conversion to a different library may fail in which
```

Comment:
there is a test [here](https://github.com/scikit-learn/scikit-learn/pull/27961/files#diff-d0cf1401f90f9cff12a7dd9778bafb5d1a06137f4145be07c8530e2502af131fR673) to cover those lines (I checked it does locally) but only when both torch and array-api-strict are installed, is there such a configuration in one of the CI jobs?